cmd/install.go

```diff
@@ -524,8 +524,35 @@ func checkInstalledPackages(ctx context.Context, tool constants.ToolType) (err e
 		}
 	}
 
-	if len(notInstall) > 0 {
-		return errors.New(strings.Join(notInstall, ", ") + " is missing from your OS, which is required for running, please install them")
+	if len(notInstall) == 0 {
+		return nil
+	}
+
+	pkgsStr := strings.Join(notInstall, ",")
+	// TODO: devise a mechanism for installing pkgs for mac & windows
+	if utils.GetOS() != constants.Linux {
+		log.WithContext(ctx).WithField("missing-packages", pkgsStr).
+			WithError(errors.New("missing required pkgs")).
+			Error("automatic install for required pkgs only set up for linux")
+
+		return fmt.Errorf("missing required pkgs: %s", pkgsStr)
+	}
+
+	return installMissingReqPackagesLinux(notInstall)
+}
+
+func installMissingReqPackagesLinux(pkgs []string) error {
+	log.WithField("packages", strings.Join(pkgs, ",")).
+		Info("system will now install missing packages")
+
+	for _, pkg := range pkgs {
+		out, err := RunCMD("sudo", "apt", "install", "-y", pkg)
```

Comment:
can log pkg we're trying to install?

Reply:
already logged above. 
if any pkg fails, it will display a proper log message

Reply:
you can add pkg names to 		Info("system will now install missing packages")